src/compiler/compile/Component.ts

```diff
@@ -793,20 +793,29 @@ export default class Component {
 				}
 
 				let deep = false;
-				let names: string[] | undefined; 
-
-				if (node.type === 'AssignmentExpression') {
+				let names: string[] = []; 
+
+				if (node.type === 'AssignmentExpression' &&
+					node.left.type === 'ArrayPattern') {
+					walk(node.left, {
+						enter(node: Node, parent: Node) {
+							if (node.type === 'Identifier' &&
+								parent.type !== 'MemberExpression') {
+									names.push(node.name);
+							}
+						}
+					});
+				} else if (node.type === 'AssignmentExpression') {
 					deep = node.left.type === 'MemberExpression';
 					names = deep
 						? [get_object(node.left).name]
 						: extract_names(node.left);
```

Comment:
This is a small point, but We can commonize `node.type === 'AssignmentExpression'`.

```ts
if (node.type === 'AssignmentExpression') {
	if (node.left.type === 'ArrayPattern') {
		walk(node.left, {
			enter(node: Node, parent: Node) {
				if (node.type === 'Identifier' && parent.type !== 'MemberExpression') {
					names.push(node.name);
				}
			}
		});
	} else {
		deep = node.left.type === 'MemberExpression';
		names = deep
			? [get_object(node.left).name]
			: extract_names(node.left);
		}
```

Reply:
updated: https://github.com/sveltejs/svelte/pull/7966/commits/28565cfa9947a2db40bf29fdd3f015a6fda863ee